src/H5Fint.c

```diff
@@ -1287,6 +1295,25 @@ H5F__new(H5F_shared_t *shared, unsigned flags, hid_t fcpl_id, hid_t fapl_id, H5F
         if (H5P_get(plist, H5F_ACS_OBJECT_FLUSH_CB_NAME, &(f->shared->object_flush)) < 0)
             HGOTO_ERROR(H5E_FILE, H5E_CANTGET, NULL, "can't get object flush cb info")
 
+        /* Get VFD SWMR configuration */
+        if (H5P_get(plist, H5F_ACS_VFD_SWMR_CONFIG_NAME, &(f->shared->vfd_swmr_config)) < 0)
+            HGOTO_ERROR(H5E_PLIST, H5E_CANTGET, NULL, "can't get VFD SWMR config info")
+
+        /* Initialization for VFD SWMR */
```

Comment:
All of the fields for VFD SWMR should be in a sub-struct, to make it clear what they are applying to.

Reply:
Additionally, this struct is 'calloc'd, and only the non-zero fields need to be initialized